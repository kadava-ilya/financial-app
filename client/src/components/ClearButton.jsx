import React from "react";
import {IconButton} from "@mui/material";
import ClearInputIcon from "../assets/icons/ClearInputIcon";
import {useDispatch} from "react-redux";

export const ClearButton = ({onClick}) => {

    const dispatch = useDispatch();

    return (
        <IconButton
            sx={{padding: 0}}
            onClick={() => dispatch(onClick)}
        >
            <ClearInputIcon />
        </IconButton>
    )
}