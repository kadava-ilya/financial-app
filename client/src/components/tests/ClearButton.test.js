import {ClearButton} from "../ClearButton";
import {fireEvent, render, screen} from "@testing-library/react";
import * as reduxHooks from "react-redux";

jest.mock('react-redux')

const mockedDispatch = jest.spyOn(reduxHooks, 'useDispatch')

describe('ClearButton component', () => {
    it('should create a ClearButton component', () => {
        const component = render(<ClearButton />)

        expect(component).toMatchSnapshot()
    })

    it('should dispatch actions on ClearButton click', () => {
        const dispatch = jest.fn()
        mockedDispatch.mockReturnValue(dispatch);

        render(<ClearButton />)

        fireEvent.click(screen.getByRole('button'))

        expect(dispatch).toHaveBeenCalled()
    })
})
