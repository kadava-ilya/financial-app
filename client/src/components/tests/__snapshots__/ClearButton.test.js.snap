// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClearButton component should create a ClearButton component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-av538e-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          enable-background="new 0 0 64 64"
          height="30px"
          id="Layer_1"
          version="1.1"
          viewBox="0 0 64 64"
          width="30px"
          x="0px"
          xml:space="preserve"
          xmlns="http://www.w3.org/2000/svg"
          xmlns:xlink="http://www.w3.org/1999/xlink"
          y="0px"
        >
          <g>
            <line
              fill="none"
              stroke="#000000"
              stroke-miterlimit="10"
              stroke-width="2"
              x1="18.947"
              x2="45.045"
              y1="17.153"
              y2="43.056"
            />
          </g>
          <g>
            <line
              fill="none"
              stroke="#000000"
              stroke-miterlimit="10"
              stroke-width="2"
              x1="19.045"
              x2="44.947"
              y1="43.153"
              y2="17.056"
            />
          </g>
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </body>,
  "container": <div>
    <button
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-av538e-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
    >
      <svg
        enable-background="new 0 0 64 64"
        height="30px"
        id="Layer_1"
        version="1.1"
        viewBox="0 0 64 64"
        width="30px"
        x="0px"
        xml:space="preserve"
        xmlns="http://www.w3.org/2000/svg"
        xmlns:xlink="http://www.w3.org/1999/xlink"
        y="0px"
      >
        <g>
          <line
            fill="none"
            stroke="#000000"
            stroke-miterlimit="10"
            stroke-width="2"
            x1="18.947"
            x2="45.045"
            y1="17.153"
            y2="43.056"
          />
        </g>
        <g>
          <line
            fill="none"
            stroke="#000000"
            stroke-miterlimit="10"
            stroke-width="2"
            x1="19.045"
            x2="44.947"
            y1="43.153"
            y2="17.056"
          />
        </g>
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
